### PR TITLE
DROOLS-2420: [DMN Designer] Hit Policy Collect does not need an Aggregation function

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-2420

Basically, when the ```HitPolicy``` is ```COLLECT``` you don't need to specify a ```BuiltinAggregator```.

### CODE REVIEW

**Comment by jomarko** [comment]
jenkins please retest this

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/hitpolicy/HitPolicyEditorViewImpl.java [hunk 1]
```diff
@@ -58,11 +57,11 @@ public HitPolicyEditorViewImpl() {
     public HitPolicyEditorViewImpl(final Select lstHitPolicies,
                                    final Select lstBuiltinAggregator,
                                    final Select lstDecisionTableOrientation,
-                                   final Document document) {
```
[the remark this answers is not in the document]
Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/hitpolicy/HitPolicyEditorViewImpl.java]: @jomarko I've merged your #1532 so will resolve the conflict here.

I also like you idea of showing "None" when the Hi Policy is not COLLECT so will make the change.

[the remark this answers is not in the document]
Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/dtable/hitpolicy/HitPolicyEditorViewImpl.java]: Thanks, just whistle when ready for re-review.

**Comment by manstis** [comment]
@jomarko This is ready for re-review... I've changed it so when you set the ```HitPolicy``` it sets the ```BuiltinAggregator``` to ```null```. This ensures the UI shows "<None>" whenever the ```HitPolicy``` is set. It also means the default ```BuiltinAggregator``` is "<None>" when you change the ```HitPolicy``` to ```COLLECT``` which, IMO, is better too.

**Comment by manstis** [comment]
Jenkins please retest this.

**Pull request merged**